plugins/nf-azure/src/main/nextflow/cloud/azure/fusion/AzFusionEnv.groovy

```diff
@@ -49,17 +49,26 @@ class AzFusionEnv implements FusionEnv {
             throw new IllegalArgumentException("Missing Azure Storage account name")
         }
 
-        if (cfg.storage().accountKey && cfg.storage().sasToken) {
-            throw new IllegalArgumentException("Azure Storage Access key and SAS token detected. Only one is allowed")
+        result.AZURE_STORAGE_ACCOUNT = cfg.storage().accountName
+
+        if (cfg.activeDirectory().isConfigured()) {
+            result.AZURE_TENANT_ID = cfg.activeDirectory().tenantId
+            result.AZURE_CLIENT_ID = cfg.activeDirectory().servicePrincipalId
+            result.AZURE_CLIENT_SECRET = cfg.activeDirectory().servicePrincipalSecret
+            return result
         }
 
-        result.AZURE_STORAGE_ACCOUNT = cfg.storage().accountName
-        // In theory, generating an impromptu SAS token for authentication methods other than
-        // `azure.storage.sasToken` should not be necessary, because those methods should already allow sufficient
-        // access for normal operation. Nevertheless, #5287 heavily implies that failing to do so causes the Azure
-        // Storage plugin or Fusion to fail. In any case, it may be possible to remove this in the future.
-        result.AZURE_STORAGE_SAS_TOKEN = getOrCreateSasToken()
+        if (cfg.managedIdentity().isConfigured()) {
+            // User-assigned Managed Identity
+            if (cfg.managedIdentity().clientId) {
+                result.AZURE_CLIENT_ID = cfg.managedIdentity().clientId
+            }
+            // System Managed Identity
+            return result
```

Comment:
> See here for an explanation on how to do this with azcopy (non Fusion): https://github.com/nextflow-io/nextflow/issues/3314#issuecomment-1836300135

This approach would work for a system-wide managed identity, but AFAIK `azcopy` requires the `AZURE_CLIENT_ID` environment variable for a user-assigned managed identity (or the `AZURE_CLIENT_ID`, `AZURE_SECRET_ID`, and `AZURE_TENANT_ID` for a service principal). 

I'm sure I'm missing something, but since Fusion support for this (wip) uses the same underlying mechanism (`DefaultAzureCredential`), it requires the same environment variables, so I'm not sure what we gain here?

Reply:
> I'm not sure we can guarantee the Azure worker nodes will have the same managed identity as the Nextflow head node. In AWS we use the configuration item `aws.batch.jobRole = 'JOB ROLE ARN'` to specify this variable, perhaps we should do the same here?

No idea about this, sorry.